ARCs/arc-0041.md

```diff
@@ -0,0 +1,107 @@
+---
+arc: 41
+title: ASA's vault of ASAs
+description: Application ID storing ASAs for another ASA
+author: Stéphane BARROSO (@SudoWeezy)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/213
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-06-16
+requires: 3, 16
+---
+
+## Abstract
+This standard is inspired by <a href="https://eips.ethereum.org/EIPS/eip-6551">EIP-6551</a>
+The goal is to allow an ASA to detain & manage ASAs or Algo through a ”vault”.
+
+## Specification
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+### Glossary
+In the document we will use following syntax:
+- Main ASA -> ASA that detain & manage others ASAs
+- Vault -> Application ID holding ASAs of the Main ASA
+
+### Metadata field
+In order to allow wallet to identify the vault of the Main ASA must be specified in the metadata.
+Using [ARC-16](./arc-0016.md):
+```json
+{
+  "title": "Main ASA",
+  "type": "object",
+  "properties": {
+    "vault_id": {
+      "type": "integer",
+      "description": "Application ID holding ASAs of the main ASA"
+    }
+  }
+}
+```
+
+```json
+{
+  "name": "Main ASA",
+  "description": "I have control over other ASAs through my vault",
+  "properties": {
+    "vault_id": 123456789
+  }
+}
+```
+### Interface
+```json
+{
+  "name": "ARC-41",
+  "desc": "Main ASA's Vault Base Interface",
+  "methods": [
+    {
+      "name": "isOwner",
+      "desc": "Indicate if the caller detain the Main ASA of this vault, this method MUST be called before calling every other method of the application",
```

Comment:
The wording make it seem like the caller must always send two app call txns. Instead, the implementer can simply add call the `isOwner` logic internally when calling any of the other methods. I think this should be made more clear.